### PR TITLE
Add ERC: Agent Offchain Conditional Settlement

### PR DESCRIPTION
**Pull request by xrqin**
This ERC defines a minimal interoperability layer for agent-native off-chain conditional settlement over
  existing off-chain channel frameworks.

  - EIP-712 typed data for proof-bound conditional obligations (`ConditionalLock`, SettlementProofRef,
  `ClaimRelayRequest`)
  - Minimal on-chain extension interface (`settleConditional`, refundConditional, `lockStatus`)
  - Canonical identifiers for condition types (HTLC, ZK_PROOF, ORACLE_ATTESTATION, etc.) and proof types
  (RECEIPT_ROOT, ZK_PROOF, RELAY_CLAIM, etc.) with open keccak256 registry convention
  - Optional extensions for capability discovery, private claim relay, and local rebalancing coordination

  ## Motivation

  Autonomous agents are natural state channel participants — always online, can sign automatically, and benefit
  from liveness requirements that humans rejected. Existing agent-related ERCs focus on
  identity/accounts/permissions but lack a common settlement layer for high-frequency machine-to-machine
  interactions.

  This ERC standardizes the settlement envelope without dictating the full channel stack. It composes with
  existing host frameworks (lifecycle, custody, dispute) rather than replacing them.

  ## Design Decisions

  - Settlement layer only: lifecycle and custody are explicitly delegated to the host framework
  - Standalone companion: designed to compose with frameworks like ERC-7824 without formal coupling
  - Proof-system agnostic: proofType identifies the verification pathway, not the proof internals
  - Open registry: canonical identifiers use keccak256 over human-readable labels; third-party extensions
  use keccak256("NAMESPACE.TYPE") without requiring ERC amendments

  ## Open Questions (for discussion)

  - Should capability discovery be part of the core standard or remain an extension?
  - Should the relationship to application protocols such as ERC-8183 be made more explicit?
  - Should a future ERC define a canonical JSON schema for capabilityURI?
  - Should relay fee settlement be standardized further?
  - Should local rebalancing coordination be split into a dedicated ERC?

  ## Checklist

  - [x] Follows ERC template format
  - [x] Includes Abstract, Motivation, Specification, Rationale, Backwards Compatibility, Security
  Considerations
  - [ ] Test vectors (planned for future revision)
  - [ ] Reference implementation (planned for future revision)

  Discussion thread: TBD (will update after creating Ethereum Magicians post)

### CODE REVIEW

File: ERCS/erc-8200.md [hunk 1]
```diff
@@ -0,0 +1,771 @@
+---
+eip: 8200
```
Comment (abcoathup) [inline ERCS/erc-8200.md]:
```suggestion
eip: 8203
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.

File: ERCS/erc-8203.md [hunk 2]
```diff
@@ -0,0 +1,771 @@
+---
+eip: 8200
+title: Agent Conditional Settlement Interface
```
Comment (abcoathup) [inline ERCS/erc-8203.md]:
```suggestion
title: Agent Offchain Conditional Settlement
```
Note: I am NOT an editor.

Interface is superfluous for an ERC.  Would offchain be worth adding to the title?

File: ERCS/erc-8200.md [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+title: Agent Conditional Settlement Interface
+description: Typed data and extension interfaces for agent-native off-chain conditional settlement over channel frameworks.
+author: Xianrui Qin (@xrqin)
+discussions-to: https://ethereum-magicians.org/t/draft-erc-agent-off-chain-conditional-settlement-extension-interface/28041
```
Comment (abcoathup) [inline ERCS/erc-8200.md]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/erc-8203-agent-off-chain-conditional-settlement-extension-interface/28041
```
Added assigned number to Eth Magicians title

**Comment by Aboudjem** [comment]
ref impl for ERC-8203 is up: https://github.com/Aboudjem/erc-8203-ref

covers the march 25 spec revision (orthogonal conditionType/proofType taxonomy, mandatory host-state binding, ERC-5267, all 10 changes). 24 tests, CI green

File: ERCS/erc-8203.md [hunk 4]
```diff
@@ -0,0 +1,802 @@
...
+type: Standards Track
+category: ERC
+created: 2026-03-10
+requires: 165, 712, 1271, 5267
+---
+
+## Abstract
+
+This ERC defines a minimal interoperability layer for agent-native off-chain conditional settlement.
+The protocol itself is actor-agnostic; agents are the primary motivating user class, but nothing prevents human signers or other automated systems from using the same interfaces.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:17]: If it isn't specific to agents, why mention agents at all? The more generally applicable a proposal, the better.

File: ERCS/erc-8203.md [hunk 5]
```diff
@@ -0,0 +1,802 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2026-03-10
+requires: 165, 712, 1271, 5267
+---
+
+## Abstract
+
+This ERC defines a minimal interoperability layer for agent-native off-chain conditional settlement.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:16]: Good, now explain what agent-native off-chain conditional settlement actually is. Remember that not everyone reading your proposal will be as intimately familiar with your domain as you are.

File: ERCS/erc-8203.md [hunk 6]
```diff
@@ -0,0 +1,802 @@
...
+The protocol itself is actor-agnostic; agents are the primary motivating user class, but nothing prevents human signers or other automated systems from using the same interfaces.
+
+It standardizes:
+
+- [EIP-712](./eip-712.md) typed data for proof-bound conditional obligations and related settlement intents.
+- A minimal on-chain conditional settlement extension interface for settlement, refund, and lock-status queries.
+- Optional extensions for capability discovery and private claim relay.
+
+This ERC is designed to compose with existing off-chain channel frameworks rather than replace them.
+It does **not** standardize channel lifecycle primitives, custody or exit primitives, routing algorithms, rebalancing algorithms, application-layer job or commerce state machines, specific zero-knowledge circuits, privacy pools, liquidity management strategies, or hub business logic.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:26]: This content, while interesting, isn't necessarily good content for your abstract. It could go in motivation or rationale instead. The abstract should give the reader just enough information to understand what problem your proposal is solving and a high-level (but still technical) overview of how it solves that problem.

File: ERCS/erc-8203.md [hunk 7]
```diff
@@ -0,0 +1,802 @@
...
+### Roles
+
+- `Agent`: an autonomous signer participating in off-chain settlement.
+- `Hub`: a coordinating counterparty or routing service that may aggregate or forward settlement.
+- `Relay`: an optional third party that submits a claim or settlement transaction on behalf of an agent.
+- `Host Framework`: an existing off-chain channel or settlement framework that defines lifecycle and custody primitives.
+- `Adjudicator`: an on-chain contract that exposes conditional settlement extension hooks and may be composed with a host-framework adjudicator.
+- `Verifier`: an on-chain contract that implements verification logic for a given proof type.
+
+Contracts implementing the core interface or any extension defined in this ERC SHOULD implement [ERC-165](./eip-165.md) and return `true` for the relevant interface identifiers.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:82]: You're kinda hiding a requirement in your definitions section. I'd move this down with the rest of the other requirements.

File: ERCS/erc-8203.md [hunk 8]
```diff
@@ -0,0 +1,802 @@
...
+### Design Goals
+
+An implementation conforming to this ERC SHOULD support:
+
+- Near-zero-gas happy-path settlement through off-chain signatures.
+- Conditional settlement triggered by proofs, attestations, or timeouts.
+- Strong replay protection across channels, chains, and intents.
+- Compatibility with both EOAs and contract accounts.
+- Composability with existing channel frameworks.
+- Composability with [ERC-4337](./eip-4337.md) wallets, private relays, and application-specific backends.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:93]: I'm not exactly sure I understand. If these are design goals, then they're talking about the standard itself and not the implementation right?

File: ERCS/erc-8203.md [hunk 9]
```diff
@@ -0,0 +1,802 @@
...
+
+This ERC MUST NOT be interpreted as standardizing:
+
+- The canonical lifecycle primitives for channels or sessions.
+- The canonical custody, transfer, or exit format.
+- The canonical routing path format.
+- The canonical rebalancing algorithm.
+- The canonical proof system.
+- The canonical privacy backend.
+- The economic policy of hubs.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:105]: I would remove the uppercase keyword from this section. This is more background on the spec itself, and not really something the implementation needs to adhere to.

File: ERCS/erc-8203.md [hunk 10]
```diff
@@ -0,0 +1,802 @@
...
+
+### Relationship to Application Protocols
+
+This ERC is not intended to standardize a job, order, marketplace, subscription, or evaluator-driven commerce state machine.
+
+Application-layer protocols such as [ERC-8183](./eip-8183.md) MAY define business objects, provider/evaluator roles, funding flows, submission flows, and terminal outcomes.
+
+This ERC instead standardizes the lower-layer settlement envelope for proof-bound conditional obligations that may need to be represented and resolved within or alongside an off-chain channel or settlement framework.
+
+An application protocol MAY compose with this ERC, but conformance to this ERC does not require any particular application object model.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:125]: This chunk interleaves specification with rationale. While we do permit small bits of rationale in the specification section, generally speaking large bits of explanatory text should be in rationale.

File: ERCS/erc-8203.md [hunk 11]
```diff
@@ -0,0 +1,802 @@
...
+
+- an [EIP-712](./eip-712.md) signature over `ClaimRelayRequest`,
+- an [ERC-1271](./eip-1271.md) contract signature check,
+- an implementation-specific proof tied to `escrowCommitment`.
+
+#### Relay Gas Model
+
+This ERC defines two relay gas models. An implementation MAY support either or both.
+
+**Model A: Direct Relay Settlement**
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:358]: Use a fifth level heading here.

File: ERCS/erc-8203.md [hunk 12]
```diff
@@ -0,0 +1,802 @@
...
+- Agent identity and reputation standards: Orthogonal. Standards that address who an agent is and whether it is trusted are complementary; this ERC addresses how an agent settles conditional obligations.
+- [ERC-8183](./eip-8183.md) (job escrow and evaluator lifecycle): Higher layer. [ERC-8183](./eip-8183.md) defines business objects (jobs, submissions, evaluations); this ERC defines the lower-layer settlement envelope that such a job protocol may use for conditional payment resolution.
+- Signed-voucher unidirectional micropayment channels: Narrower scope. Standards that define a specific payment topology (single-direction, signed-voucher) for AI service micropayments address a different layer. This ERC is topology-agnostic and proof-system-agnostic; it standardizes the settlement interface, not a specific channel construction.
+
+The key differentiator of this ERC is that it is a **proof-bound conditional settlement interface over an existing host framework**, not a standalone channel protocol, identity layer, or application state machine.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:710]: Comparisons to other proposals should likely live in the motivation section. That said, I'm not entirely sure this section is necessary?

File: ERCS/erc-8203.md [hunk 13]
```diff
@@ -0,0 +1,802 @@
...
+It adds a common conditional settlement extension interface above them.
+
+## Test Cases
+
+Future versions of this draft will include:
+
+- [EIP-712](./eip-712.md) hashing test vectors,
+- signature verification examples for EOAs and [ERC-1271](./eip-1271.md) accounts,
+- conditional refund edge cases,
+- proof reference examples for multiple proof families.
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:732]:
```suggestion
<!-- TODO: test cases -->
```

File: ERCS/erc-8203.md [hunk 14]
```diff
@@ -0,0 +1,802 @@
...
+
+## Reference Implementation
+
+See [`IAgentConditionalSettlementExtension.sol`](../assets/eip-8203/IAgentConditionalSettlementExtension.sol).
+
+## Security Considerations
+
+### Stale State And Replay
+
+Implementations MUST protect against:
```
Comment (SamWilsn) [inline ERCS/erc-8203.md:742]: Keep requirements (defined with UPPERCASE keywords) in the specification section. Discuss their security implications here.
